ap2/utils.py

```diff
@@ -79,7 +79,10 @@ def get_pycaw_volume_session():
 def get_volume():
     subsys = platform.system()
     if subsys == "Darwin":
-        pct = int(subprocess.check_output(["osascript", "-e", "output volume of (get volume settings)"]).rstrip())
+        try:
+            pct = int(subprocess.check_output(["osascript", "-e", "output volume of (get volume settings)"]).rstrip())
+        except:
+            pct = 0
```

Comment:
What are the reasons that it can't be retrieved? Is there anything that can be done as a work-around?

Reply:
That's apple script - it's a simple global generic way to get the system volume for whichever device is current set as output (often just "<device> speakers"). When I run `osascript -e '(get volume settings)'` on my sys when something else is set, I get e.g.:

`output volume:missing value, input volume:100, alert volume:100, output muted:missing value`

Reply:
On my external soundcard M-Track 2x2 the volume is fixed and it can't be retrieved with osascript or via System Preferences, it throws a value error when I'm trying to start the program using my external soundcard as the default output device